examples/motoko_examples/whoami/src/whoami.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    Canister,
+    CanisterResult,
+    ic,
+    Init,
+    PostUpgrade,
+    Principal,
+    Query,
+    UpdateAsync,
+    Variant
+} from 'azle';
+
+type WhoAmICanister = Canister<{
+    installer(): CanisterResult<Principal>;
+    argument(): CanisterResult<Principal>;
+    whoami(): CanisterResult<Principal>;
+    id(): CanisterResult<Principal>;
+    idQuick(): CanisterResult<Principal>;
+}>;
+
+type WhoAmIResult = Variant<{
+    ok: Principal;
+    err: string;
+}>;
+
+let install: Principal;
+let someone: Principal;
+
+// Manually save the calling principal and argument for later access.
+export function init(somebody: Principal): Init {
+    install = ic.caller();
+    someone = somebody;
+}
+
+// Manually re-save these variables after new deploys.
+export function postUpgrade(somebody: Principal): PostUpgrade {
+    install = ic.caller();
+    someone = somebody;
+}
+
+// Return the principal identifier of the wallet canister that installed this
+// canister.
+export function installer(): Query<Principal> {
```

Comment:
We might want these functions to return `Opt<Principal>`. The typing is slightly different than Motoko because of the Azle environment, we don't know that init and postUpgrade will set these values so we kind of need the `Opt` type.